src/main/java/roomescape/service/ReservationService.java

```diff
@@ -0,0 +1,40 @@
+package roomescape.service;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import org.springframework.stereotype.Service;
+import roomescape.domain.Reservation;
+import roomescape.dto.ReservationRequest;
+import roomescape.dto.ReservationResponse;
+import roomescape.exception.NotFoundException;
+
+@Service
+public class ReservationService {
+
+    private final List<Reservation> reservations = new ArrayList<>();
+    private final AtomicLong index = new AtomicLong(0);
+
+    public List<Reservation> getAllReservations() {
+        return reservations;
+    }
+
+    public ReservationResponse addReservation(ReservationRequest request) {
+        Long newId = index.incrementAndGet();
+        Reservation reservation = request.toReservation(newId);
+        reservations.add(reservation);
+        return new ReservationResponse(
+                reservation.getId(),
+                reservation.getName(),
+                reservation.getDate(),
+                reservation.getTime()
+        );
+    }
+
+    public void deleteReservation(Long id) {
+        boolean removed = reservations.removeIf(reservation -> reservation.getId().equals(id));
+        if (!removed) {
```

Comment:
```suggestion
        boolean isDeleted = reservations.removeIf(reservation -> reservation.getId().equals(id));
        if (!isDeleted) {
```
변수명을 수정해 보면 어떨까요? is/has 로 시작하는 것이 자연스럽고, 일반적이라서요!